src/trio/_tests/test_socket.py

```diff
@@ -1116,7 +1132,8 @@ async def test_many_sockets() -> None:
         try:
             a, b = stdlib_socket.socketpair()
         except OSError as e:  # pragma: no cover
-            assert e.errno in (errno.EMFILE, errno.ENFILE)
+            # the noqa is "Found assertion on exception `e` in `except` block"
+            assert e.errno in (errno.EMFILE, errno.ENFILE)  # noqa: PT017
```

Comment:
.

Reply:
This one is quite interesting actually. This one is a loop that breaks prematurely only if the kernel responds with an `Out of socket file ids` error.

Reply:
hmm, this is a weird test. it loops `_x` until it gets `total//2-1` unless it throws an exception - ~~in which case it'd break and then the test would fail later in the last~~ nowait that's just a print! That print should totally be inside the `except` catcher and `_x` can be renamed to `_`. Probably also a `pragma: no cover`? codecov is acting weird so idk if it's getting covered or not.

Okay, so, it loops until it fails to create a socket - or it hits `total//2`. So supposedly it's not an error if it raises earlier, nor is it a problem if it doesn't raise, which means that `pytest.raises` is not relevant.

It'd maybe be nice to add some comments. Why `pytest.raises` isn't relevant, and maybe also nice to add that EMFILE is "Too many open files" and ENFILE is "File table overflow".

Reply:
Fixed I think with 48b771b4